### PR TITLE
[IMX] compilation error fix & others

### PR DESCRIPTION
**Pull request by mk01**

### CODE REVIEW

**Comment by FernetMenta** [comment]
that's IMX code only, right?

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp [hunk 1]
```diff
@@ -1517,18 +1517,15 @@ int setIPUMotion(bool hasPrev, EINTERLACEMETHOD imethod)
 void CIMXContext::Blit(CIMXBuffer *source_p, CIMXBuffer *source, const CRect &srcRect,
                        const CRect &dstRect, uint8_t fieldFmt, int page)
 {
+  static unsigned char pg;
```
Comment (vpeter4) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp]: uninitialized variable?

**Comment by fritsch** [comment]
@mk01 everything in that you want to have in - are there commits that touch something != IMX code paths?

**Comment by fritsch** [comment]
@mk01 Everything in you need? If you feel ready, just ping me and we shovle it in.

**Comment by MartijnKaijser** [comment]
@mk01 situation update?

**Comment by MartijnKaijser** [comment]
@mk01 can this be merged?

**Comment by jerome-benoit** [comment]
This patches set is successfully used by the LibreELEC project for the i.MX6 port.

**Comment by fritsch** [comment]
@mk01 please squash all of them the way you wanted to bisect them if something goes wrong.

**Comment by fritsch** [comment]
@mk01 talk with us :-) that code here won't be maintainable by anyone else than exactly you. So, what's your goal here? We need to know when this will be ready and about your plans.

**Comment by vpeter4** [comment]
@mk01: With this PR I'm getting immediate system freeze when video starts on Udoo Quad with LVDS display attached. With HDMI TV everything is ok on all "mine" imx6 systems. Any tip what to look for? Obviously there is no log possible.

**Comment by mk01** [comment]
superseded by https://github.com/xbmc/xbmc/pull/11297

**Pull request closed without merging**